### PR TITLE
Core/Packets: SMSG_NOTIFICATION

### PR DESCRIPTION
**Pull request by Golrag**

### CODE REVIEW

File: src/server/game/Server/Packets/ChatPackets.h [hunk 1]
```diff
@@ -196,6 +196,17 @@ namespace WorldPackets
             int32 SoundIndex = -1;
             int32 EmoteID = 0;
         };
+
+        class PrintNotification final : public ServerPacket
+        {
+        public:
+            PrintNotification() : ServerPacket(SMSG_NOTIFICATION, 3) { }
+            PrintNotification(std::string notifyText) : ServerPacket(SMSG_NOTIFICATION, 3), NotifyText(notifyText) { }
```
Comment (DDuarte) [inline src/server/game/Server/Packets/ChatPackets.h]: std::string const&

Comment (Golrag) [inline src/server/game/Server/Packets/ChatPackets.h]: Updated ! :)

**Comment by DDuarte** [comment]
Thanks!

Btw, anyone that doesn't yet believe in our 'packet structures' should look at this. The code to send a packet got simplified \*a lot\*

File: src/server/scripts/Commands/cs_message.cpp [hunk 2]
```diff
@@ -172,9 +170,7 @@ class message_commandscript : public CommandScript
         std::string str = handler->GetTrinityString(LANG_GM_NOTIFY);
         str += args;
 
-        WorldPacket data(SMSG_NOTIFICATION, (str.size()+1));
-        data << str;
-        sWorld->SendGlobalGMMessage(&data);
+        sWorld->SendGlobalMessage(WorldPackets::Chat::PrintNotification(str).Write());
```
Comment (joschiwald) [inline src/server/scripts/Commands/cs_message.cpp]: why change SendGlobalGMMessage to SendGlobalMessage?

**Comment by AwkwardDev** [comment]
@DDuarte, I think everyone who knows what's going on can only like it :+1:

File: src/server/scripts/Commands/cs_message.cpp (hunk 2 above)
Comment (Golrag) [inline src/server/scripts/Commands/cs_message.cpp]: Copy/paste fail I guess, Updating x')

**Pull request merged**